src/dag/read.rs

```diff
@@ -63,30 +63,76 @@ mod tests {
                 .await
                 .unwrap();
             kvw.commit().await.unwrap();
             assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
 
-            // Mayve test isolation: other txs should not affect what the read tx, which is
+            // Maybe test isolation: other txs should not affect what the read tx, which is
             // still open, sees.
-            if del_haskey {
+            if del_key {
                 let kvw2 = kv.write().await.unwrap();
-                kvw2.del("haskey".into()).await.unwrap();
+                kvw2.del(k).await.unwrap();
                 kvw2.commit().await.unwrap();
-                assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
             }
+            assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
         }
 
-        test("haskey", false, true).await;
-        test("haskey", true, true).await;
-        test("missing", false, false).await;
+        test("present", false, true).await;
+        test("present", true, true).await;
+        test("no such hash", false, false).await;
+    }
+
+    #[async_std::test]
+    async fn test_get_chunk() {
+        async fn test(data: Vec<u8>, refs: &[&str], get_same_chunk: bool) {
+            let kv = MemStore::new();
+            let cloned_data = data.clone();
+            let kvw = kv.write().await.unwrap();
+            let chunk = Chunk::new((data, 0), refs);
+            kvw.put(&Key::ChunkData(&chunk.hash()).to_string(), chunk.data())
+                .await
+                .unwrap();
+            if let Some(meta) = chunk.meta() {
+                kvw.put(&Key::ChunkMeta(chunk.hash()).to_string(), meta)
+                    .await
+                    .unwrap();
+            }
+            kvw.commit().await.unwrap();
+
+            // TODO(phritz) start a write tx here that completes before r.get_chunk
+            // below and ensure it does not affect the result.
+            let kvr = kv.read().await.unwrap();
+            let r = Read { kvr };
+            // TODO(phritz) start a write tx here that completes before r.get_chunk
+            // and ensure it does not affect the result.
+            if get_same_chunk {
+                let got_option = r.get_chunk(chunk.hash()).await.unwrap();
+                let got_chunk = got_option.unwrap();
+                assert_eq!(chunk.hash(), got_chunk.hash());
+                let d: &[u8] = &cloned_data;
+                assert_eq!(d, got_chunk.data());
+                if chunk.refs().is_some() {
+                    assert_eq!(
+                        refs,
+                        got_chunk.refs().unwrap().collect::<Vec<&str>>().as_slice()
+                    );
+                }
+            } else {
+                let got_option = r.get_chunk("no such hash").await.unwrap();
+                assert_eq!(None, got_option);
+            }
+        }
+        test(vec![1u8], &vec!["r1", "r2"], true).await;
```

Comment:
You can say `vec![1]` (the compiler knows it must be u8 due to parameter type of `test`).